### PR TITLE
Apply contribution guidelines to "Attribute" rule

### PR DESCRIPTION
**Pull request by batusa**

### CODE REVIEW

File: library/Rules/Attribute.php [hunk 1]
```diff
@@ -17,7 +17,13 @@
 use Respect\Validation\Exceptions\ComponentException;
 use Respect\Validation\Validatable;
 
-class Attribute extends AbstractRelated
+/**
+ * Validates an object attribute, event private ones.
+ *
+ * @author Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ * @author Henrique Moody <henriquemoody@gmail.com>
```
Comment (henriquemoody) [inline library/Rules/Attribute.php]: You're missing your name here.

Also, could you write the docblocks for the methods in the class?

File: library/Exceptions/KeyNestedException.php [hunk 2]
```diff
@@ -13,8 +13,16 @@
 
 namespace Respect\Validation\Exceptions;
 
-class KeyNestedException extends AttributeException implements NonOmissibleException
+/**
+ * Exceptions to be thrown by the Attribute Rule.
+ *
+ * @author Henrique Moody <henriquemoody@gmail.com>
```
Comment (henriquemoody) [inline library/Exceptions/KeyNestedException.php]: You missed your name here

File: library/Exceptions/KeyException.php [hunk 3]
```diff
@@ -25,4 +33,9 @@ class KeyException extends AttributeException implements NonOmissibleException
             self::INVALID => 'Key {{name}} must not be valid',
         ],
     ];
+
+    protected function chooseTemplate(): string
```
Comment (henriquemoody) [inline library/Exceptions/KeyException.php]: missing docblock with `{@inheritdoc}`

File: library/Exceptions/AttributeException.php [hunk 4]
```diff
@@ -13,10 +13,15 @@
 
 namespace Respect\Validation\Exceptions;
 
-class AttributeException extends NestedValidationException implements NonOmissibleException
+/**
+ * Exceptions to be thrown by the Attribute Rule.
+ *
+ * @author Henrique Moody <henriquemoody@gmail.com>
```
Comment (henriquemoody) [inline library/Exceptions/AttributeException.php]: Your name should be here

File: docs/Attribute.md [hunk 5]
```diff
@@ -4,7 +4,7 @@
 - `Attribute(string $name, Validatable $rule)`
 - `Attribute(string $name, Validatable $rule, bool $mandatory)`
 
-Validates an object attribute.
+Validates an object attribute, even private ones.
```
Comment (henriquemoody) [inline docs/Attribute.md]: Nice one!

File: library/Exceptions/AttributeException.php [hunk 6]
```diff
@@ -13,10 +13,15 @@
...
+ */
+final class AttributeException extends NestedValidationException implements NonOmissibleException
 {
-    public const NOT_PRESENT = 'not_present';
-    public const INVALID = 'invalid';
+    private const NOT_PRESENT = 'not_present';
+    private const INVALID = 'invalid';
```
Comment (henriquemoody) [inline library/Exceptions/AttributeException.php]: Shall we really do it? I think we should keep them public, at least for now. 

Also I think that later we will probably set these values from the validation

File: library/Exceptions/KeyException.php [hunk 7]
```diff
@@ -13,8 +13,16 @@
...
-class KeyException extends AttributeException implements NonOmissibleException
+/**
+ * Exceptions to be thrown by the Attribute Rule.
+ *
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ */
+final class KeyException extends NestedValidationException implements NonOmissibleException
 {
+    private const NOT_PRESENT = 'not_present';
+    private const INVALID = 'invalid';
```
Comment (henriquemoody) [inline library/Exceptions/KeyException.php]: Same here

File: library/Exceptions/KeyNestedException.php [hunk 8]
```diff
@@ -25,4 +33,9 @@ class KeyNestedException extends AttributeException implements NonOmissibleExcep
             self::INVALID => 'Key chain {{name}} must not be valid',
         ],
     ];
+
+    protected function chooseTemplate(): string
```
Comment (henriquemoody) [inline library/Exceptions/KeyNestedException.php]: Missing docblock with `{@inheritdoc}`

File: library/Rules/AbstractRelated.php [hunk 9]
```diff
@@ -94,4 +87,11 @@ public function validate($input): bool
 
         return $this->decision('validate', $hasReference, $input);
     }
+
+    private function decision($type, $hasReference, $input)
```
Comment (henriquemoody) [inline library/Rules/AbstractRelated.php]: Nice!

**Comment by batusa** [comment]
@henriquemoody ready for review!

**Pull request merged**